src/sys/unix/selector/epoll.rs

```diff
@@ -133,6 +126,43 @@ impl Drop for Selector {
     }
 }
 
+/// Creates a new epoll file descriptor with close-on-exec flag set.
+///
+/// close-on-exec is set atomically with `epoll_create1()` if possible,
+/// otherwise `epoll_create()` and `fcntl()` calls are used as a fallback.
+fn new_epoll_fd() -> io::Result<libc::c_int> {
+    // According to libuv, `EPOLL_CLOEXEC` is not defined on Android API <
+    // 21. But `EPOLL_CLOEXEC` is an alias for `O_CLOEXEC` on that platform,
+    // so we use it instead.
+    #[cfg(target_os = "android")]
+    let flag = libc::O_CLOEXEC;
+    #[cfg(not(target_os = "android"))]
+    let flag = libc::EPOLL_CLOEXEC;
+
+    // Illumos and Redox have epoll_create1() function, but no epoll syscalls.
+    #[cfg(any(target_os = "illumos", target_os = "redox"))]
+    let ep = syscall!(epoll_create1(flag))?;
+
+    // Try epoll_create1 syscall with an epoll_create fallback.
+    #[cfg(not(any(target_os = "illumos", target_os = "redox")))]
```

Comment:
Could you change this to be android only with a comment referring to https://github.com/tokio-rs/mio/issues/1473 and explaining that Android API level 16 doesn't define `epoll_create1`. I really like to minimise the use of raw system calls.